libs/SalesforceSDKCore/SalesforceSDKCore/Classes/PushNotification/SFSDKPushNotificationDecryption.m

```diff
@@ -199,7 +199,7 @@ + (SFEncryptionKey *)getAESKeyFromSecret:(NSString *)secret error:(NSError **)er
     }
 
     CFRelease(privateKeyRef);
-    if (decryptedData == nil) {
+    if (decryptedData == nil || [decryptedData length] != 32) {
```

Comment:
From the comment in the flapping test (below), starting in 17.4 decrypting with a bad key returns data instead of nil so at the time I just changed our expected error in the test. But it looks like that data is also variable length (in my runs locally it was 72, 54, and 15). When it’s less the data length is less than 32, lines 209 or 210 crash so adding a length check before that